addon/-private/factory-for.js

```diff
@@ -2,13 +2,30 @@
   Utility to use `ApplicationInstance#factoryFor` if available (2.12+) and
   fallback to private `_lookupFactory` otherwise
 */
-export default function factoryFor(owner, type) {
-  let factory;
+class LegacyFactoryForWrapper {
+  constructor(factory) {
+    this._factory = factory;
+  }
+
+  get class() {
+    return this._factory;
+  }
+
+  create(fullName, options) {
+    if (this._factory) {
+      return this._factory.create(fullName, options);
+    }
+  }
+}
+
+export default function factoryFor(owner, fullName) {
   if (owner.factoryFor) {
-    let maybeFactory = owner.factoryFor(type);
-    factory = maybeFactory && maybeFactory.class;
```

Comment:
It is if you're not accessing props off the class, which is only available now via the class prop.  Instances should not be created from that class prop AFAIK and which led to the owner not being injected.